src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/SourceUpdaterStep.cs

```diff
@@ -69,7 +69,24 @@ public SourceUpdaterStep(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<
         private IEnumerable<DiagnosticDescriptor> GetDiagnosticDescriptorsForCodeFixer(CodeFixProvider fixer) =>
             _allAnalyzers.SelectMany(a => a.SupportedDiagnostics).Where(d => fixer.FixableDiagnosticIds.Contains(d.Id));
 
-        protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token) => Task.FromResult(context?.CurrentProject is not null && SubSteps.Any());
+        protected override async Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context?.CurrentProject is null)
+            {
+                return false;
+            }
+
+            var applicableSubsteps = 0;
+            foreach (var substep in SubSteps)
+            {
+                if (await substep.IsApplicableAsync(context, token).ConfigureAwait(false))
+                {
+                    applicableSubsteps++;
```

Comment:
Nit-pick - you could just return true here. No need to count the applicable sub-steps if we're only interested in knowing if >0 of them are true.